dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/SingleDotAssetToMapViewTransformer.java

```diff
@@ -0,0 +1,84 @@
+package com.dotmarketing.portlets.contentlet.transform;
+
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.beans.Identifier;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.languagesmanager.model.Language;
+import com.dotmarketing.util.UtilHTML;
+import com.dotmarketing.util.UtilMethods;
+import com.liferay.portal.language.LanguageUtil;
+import com.liferay.util.StringPool;
+import com.rainerhahnekamp.sneakythrow.Sneaky;
+import io.vavr.control.Try;
+
+import java.util.Map;
+
+/**
+ * Transform a single dotAsset to a map view
+ * @author jsanca
+ */
+public class SingleDotAssetToMapViewTransformer implements FieldsToMapTransformer {
+
+    private final Contentlet dotAsset;
+
+    public SingleDotAssetToMapViewTransformer(final Contentlet dotAsset) {
+        this.dotAsset = dotAsset;
+    }
+
+    @Override
+    public Map<String, Object> asMap() {
+
+        return Sneaky.sneaked(()->toMap()).get();
+    }
+
+    private Map<String, Object> toMap() throws DotSecurityException, DotDataException {
+
+        final Map<String, Object> dotAssetMap = new ContentletToMapTransformer(dotAsset).toMaps().get(0);
+        final Identifier identifier       = APILocator.getIdentifierAPI().find(dotAsset.getVersionId());
+        final String fileName = Try.of(()->dotAsset.getBinary("asset").getName()).getOrElse("unknown");
+
+        dotAssetMap.put("mimeType", APILocator.getFileAssetAPI().getMimeType(fileName));
```

Comment:
We should probably stop doing this kind of this  APILocator.getHostAPI()
If we want to convert this into a testable code these should be class attributes where we can choose to inject them from the APILocator or a mock. And have  constructor marked as @VisibleForTesting